File: app/lib/models/traveler_intro.dart
class TravelerIntro {
  final String travelerId;
  final String title;
  final String content;
  final String location;
  final String preferredMeetup;
  final List<String> travelTheme;
  final DateTime createdAt;
  final DateTime updatedAt;

  TravelerIntro({
    required this.travelerId,
    required this.title,
    required this.content,
    required this.location,
    required this.preferredMeetup,
    required this.travelTheme,
    required this.createdAt,
    required this.updatedAt,
  });

  factory TravelerIntro.fromJson(Map<String, dynamic> json) => TravelerIntro(
        travelerId: json['traveler_id'],
        title: json['title'],
        content: json['content'],
        location: json['location'],
        preferredMeetup: json['preferred_meetup'],
        travelTheme: List<String>.from(json['travel_theme']),
        createdAt: DateTime.parse(json['created_at']),
        updatedAt: DateTime.parse(json['updated_at']),
      );

  Map<String, dynamic> toJson() => {
        'traveler_id': travelerId,
        'title': title,
        'content': content,
        'location': location,
        'preferred_meetup': preferredMeetup,
        'travel_theme': travelTheme,
        'created_at': createdAt.toIso8601String(),
        'updated_at': updatedAt.toIso8601String(),
      };
}

File: functions/entity/traveler_intro.js
// traveler_intros 컬렉션 (여행객 소개글)
// 문서 ID: introId (자동 생성)
// {
//   traveler_id: string,
//   title: string,
//   content: string,
//   location: string,
//   preferred_meetup: '온' | '오프' | '둘 다',
//   travel_theme: string[],
//   created_at: timestamp,
//   updated_at: timestamp
// }

class TravelerIntro {
  constructor(data) {
    this.traveler_id = data.traveler_id;
    this.title = data.title;
    this.content = data.content;
    this.location = data.location;
    this.preferred_meetup = data.preferred_meetup;
    this.travel_theme = data.travel_theme;
    this.created_at = data.created_at;
    this.updated_at = data.updated_at;
  }
}

module.exports = TravelerIntro;
